acapy_agent/wallet/keys/manager.py

```diff
@@ -66,6 +75,20 @@ def __init__(self, session: ProfileSession):
         self.session: ProfileSession = session
         self.wallet: BaseWallet = session.inject(BaseWallet)
 
+    async def idempotent_resolve_and_store_from_kid(self, kid: str):
+        """Fetch key if exists, otherwise resolve and store it"""
+        if await self.kid_exists(kid):
+            LOGGER.debug(
+                f"Key for kid {kid} already exists in storage, will not resolve from ledger."
```

Comment:
```suggestion
                f"kid {kid} already bound in storage, will not resolve from ledger."
```

Reply:
could maybe still resolve to ensure its the right multikey value?

Reply:
still resolving would defeat the purpose of using the multikeymanager right? as then we are _always_ resolving the VM - not taking advantage of the multikeymanager's storage of kid<->multikey